### ISSUE TITLE
[KEYBOARD]: STEM Standalone - Focus MUST be returned to the Edit button when Update page is pressed or clicked

### ISSUE THREAD

**Issue by 1Copenut** [department-of-veterans-affairs/va.gov-team#12338]
# [508-defect-2](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-2)

**Feedback framework**

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Description

Some of the Update page buttons do not return focus to the Edit buttons on Step 6: Review Application. Users need a consistent experience so they won't have to re-orient or discover where their focus is when it is not returned to Edit. Screenshot attached below.

## Point of Contact

**VFS Point of Contact:** _Trevor_

## Acceptance Criteria

- [ ] Focus is returned to the Edit button when users press Update Page
- [ ] Edit button should have the yellow focus halo when users press Update Page

## Environment

* https://staging.va.gov/education/apply-for-education-benefits/application/10203/review-and-submit
* Test user 226

## WCAG or Vendor Guidance (optional)

* [Keyboard: Understanding SC 2.1.1](https://www.w3.org/TR/UNDERSTANDING-WCAG20/keyboard-operation-keyboard-operable.html)

## Screenshots or Trace Logs

![Screen Shot 2020-08-10 at 9 51 19 PM](https://user-images.githubusercontent.com/934879/89852435-7121d580-db54-11ea-977f-91964c0e33be.png)

**Comment by elder-joseph** [comment]
Hey @1Copenut, after looking through some other “review-and-submit” pages, I was able to find the same behavior was also present in the 1995 form for “Education Benefit” and potentially in other places throughout the app.

I was able to identify a fix, but it would be a platform wide fix by making a change in the 
`vets-website/src/platform/forms-system/src/js/utilities/ui/index.js`
The fix I found was changing the following seemingly redundant logic from
```
  if (el) {
    if (el.tabIndex === 0) {
      el.setAttribute('tabindex', '0');
    }
    if (el.tabIndex < 0) {
      el.setAttribute('tabindex', '-1');
    }
    el.focus(options);
  } 
```
to
```
  if (el) {
    el.focus(options);
  }
```
Tab index is being set automatically and removing that piece of logic corrects the issue as described in all the places I am aware that it is occuring.

**Comment by 1Copenut** [comment]
@elder-joseph Thank you for the heads up. I wasn't aware this behavior was happening in other places, but this definitely confirms it's a forms library behavior. I'm working with the VSP tools FE team on a global fix ( https://github.com/department-of-veterans-affairs/vets-website/pull/13825 ) that is in draft, but hopefully we get ironed out soon.

Let's use that as the source of truth for fixing this behavior, and. this ticket as an implementation of that fix. I'll add a blocked tag to this one in ZenHub and list the other ticket as the blocker.

**Comment by SebVortex** [comment]
@noahgelman There appear to be some failed tests and a error caused by timeout.
It can still technically be approved but that should probably be attended to (or context given).

**Comment by SebVortex** [comment]
@noahgelman 
It's safe to say that this won't get any traction until #4613 is sorted out?

**Comment by noahgelman** [comment]
@SebVortex Yes

**Comment by AdamFreemer** [comment]
@noahgelman -- any update on your code for the related PR 19085?

**Comment by noahgelman** [comment]
@AdamFreemer Am circling back around to this ticket later today

**Comment by AdamFreemer** [comment]
@noahgelman Any updates?

**Comment by mrblanco1** [comment]
@noahgelman - Hi Noah! Can you please provide Adam with an update on this ticket since it is part of our current sprint (ends tomorrow)? Or, if you could point us to someone else who can help, that would be awesome.  Thanks!

**Comment by ERRonat0609** [comment]
@AdamFreemer - I've just checked PR 19085 and it looks like it has passed all checks. It may just need a review. Can you look at that PR again? Perhaps all you need to do is request a review on Slack so that it can be merged. It says a review is required by vsp-design-system-fe.

**Comment by mrblanco1** [comment]
Noah working on his portion of the code. ETA 12/10.

**Comment by noahgelman** [comment]
I am currently working on this ticket and will have an update soon

**Comment by AdamFreemer** [comment]
```
Noah Gelman - VSP/Product Support - Web Accessibility [Ad Hoc]  7:23 AM
Hey, I wanted to give you an update. The work itself is done, but the problem is that it revealed a bug in one of the keyboard test cases. Other builds are failing b/c of this bug as well, so that bug needs to be worked out so this build can pass
7:23am
I'll keep you updated
```

**Comment by mrblanco1** [comment]
Hi @noahgelman - thank you for your feedback. Just wondering if you have an ETA? Thanks!

**Comment by noahgelman** [comment]
Looking at it today after my meetings

**Comment by mrblanco1** [comment]
@noahgelman - Hi! Happy Friday! Any word on this? Is there anything we can help with? Thanks!

**Comment by mrblanco1** [comment]
Hi @noahgelman - I hope you had a great time with family and friends. Just wanted to follow-up on this ticket. Please let us know if you have any questions.

**Comment by noahgelman** [comment]
I'll follow up soon.

**Comment by af-manish** [comment]
@noahgelman  Happy New Year. Any updates here?

**Comment by noahgelman** [comment]
The fix for the actual ticket of this is done, but the fix revealed a build
error that is currently being worked on. Will follow up with this again
tomorrow

**Comment by mrblanco1** [comment]
Hi @noahgelman - Any update on this issue? Do you know if the build problem was resolved? Thanks!

**Comment by mrblanco1** [comment]
Hi @noahgelman  - Any update on this issue? Do you know if the build problem was resolved? Thanks!

**Comment by noahgelman** [comment]
Not yet, I'll revisit this issue and update you later today

**Comment by mrblanco1** [comment]
@noahgelman  - Hi Noah! Can you please provide us an update on this issue? Is there anyone specific that needs to get involved? Thanks!

**Comment by mrblanco1** [comment]
This ticket has been Blocked since Oct 2021.

**Comment by noahgelman** [comment]
The PR is currently being QA'd

**Comment by noahgelman** [comment]
@mrblanco1 Can you provide an updated link for us to test?

**Comment by SebVortex** [comment]
@noahgelman The 10203 form is now at this url -> https://staging.va.gov/education/other-va-education-benefits/stem-scholarship/apply-for-scholarship-form-22-10203/introduction

**Comment by briandeconinck** [comment]
Thanks @SebVortex! Logging in using user 228, I am able to complete the form and test the keyboard issue originally flagged. Focus behavior now matches expectations:

- Focus is returned to the Edit button when users press Update Page
- Edit button has the yellow focus halo when users press Update Page

This looks good to go! cc @noahgelman

**Comment by SebVortex** [comment]
@noahgelman @briandeconinck 
Are we ready to close this issue? (Do we need UAT or need to remove any environment flags?)

**Comment by mrblanco1** [comment]
@SebVortex - We will conduct QA and EDU will conduct UAT and then we can close.
Thanks!

**Comment by noahgelman** [comment]
@SebVortex The PR still needs to be approved by a code owner. I will update the ticket when the code is merged

**Comment by hallihocine** [comment]
Ticket has been validated in staging: Focus is returned to the Edit button when users press Update Page and the Edit button has the yellow focus halo when users press Update Page.

**Comment by hallihocine** [comment]
- Test Run: 
https://dsvavsp.testrail.io/index.php?/runs/view/2885&group_by=cases:section_id&group_order=asc&group_id=3598
- Test Cases:
[(KEYBOARD)- STEM Standalone - Focus MUST be returned to the Edit button when Update page is pressed or clicked.xlsx](https://github.com/department-of-veterans-affairs/va.gov-team/files/7981995/KEYBOARD.-.STEM.Standalone.-.Focus.MUST.be.returned.to.the.Edit.button.when.Update.page.is.pressed.or.clicked.xlsx)

**Comment by SebVortex** [comment]
@noahgelman 
Has this been UAT'd yet?

**Comment by SebVortex** [comment]
Alright, I'll (finally) be closing the issue.
Nice work, everyone!

**Comment by SebVortex** [comment]
Alright, I'll (finally) be closing the issue.
Nice work, everyone!

**Comment by SebVortex** [comment]
Alright, I'll (finally) be closing the issue.
Nice work, everyone!

**Comment by mrblanco1** [comment]
QA and EDU UAT passed on 2/3/2022. Closing.
